flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/sink/shuffle/DataStatisticsFactory.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink.sink.shuffle;
+
+/**
+ * DataStatisticsFactory provides the DataStatistics definition for different mode like HASH, RANGE
+ */
+class DataStatisticsFactory<K> {
+
+  DataStatistics<K> createDataStatistics() {
```

Comment:
I am not sure if there is much value of this factory class. even if we want to keep it, this method should be called `createMapStatistics`.

Reply:
I would feel that we either need a `clean` method to the DataStatistics API, or need to abstract away the way to recreate a new one after sending it to the JobManager (at checkpoint). I do not feel that the Operator should know which type of statistics is used.

Alternatively we could use statistics which automatically keep historical data - but I would not bother with them in this phase